omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -571,7 +571,8 @@ def warn_if_cred_mismatch(opts)
     #
     def save_credentials_to_config
       credentials.legacy_credentials_hash.each do |service, creds|
-        next if service == "chef-server"
+        # Ignore secrets added by add-ons and the keys
+        next if PrivateChef[service].nil?
```

Comment:
I think any problems with that need to be cleared up before we get here, otherwise it would blow up with a NoMethodError [] on NilClass.  However, this is a good thing to double check on.